tools/hive-bigquery/bigquery_component.py

```
@@ -0,0 +1,632 @@
+                         dataset_id)
+        except api_exceptions.NotFound:
+            logger.debug("Table %s not found in %s dataset. No need to delete",
+                         table_name, dataset_id)
+
+    def check_bq_write_mode(self, mysql_component, hive_table_model,
+                            bq_table_model):
+        """Validates the bq_table_write_mode provided by user
+
+        If the mode is overwrite, drops the tracking table and deletes the
+        BigQuery table. If the mode is create, checks if the tracking table
+        and BigQuery table exist. If the mode is append, checks whether the
+        BigQuery table exists
+
+        Args:
+            mysql_component (:class:`MySQLComponent`): Instance of
+                MySQLComponent to connect to MySQL
+            hive_table_model (:class:`HiveTableModel`): Wrapper to Hive table
+                details
+            bq_table_model (:class:`BigQueryTableModel`): Wrapper to BigQuery
+                table details
+
+        Returns:
+            boolean: True if the write mode is okay to use, else False
+        """
+
+        if PropertiesReader.get('bq_table_write_mode') == "overwrite":
+            logger.debug("Deleting tracking table and BigQuery table...")
+            mysql_component.drop_table(hive_table_model.tracking_table_name)
+            self.delete_table(bq_table_model.dataset_id,
+                              bq_table_model.table_name)
+            hive_table_model.is_first_run = True
+
+        elif PropertiesReader.get('bq_table_write_mode') == "create":
+            if not hive_table_model.is_first_run:
+                print_and_log("Tracking Table {} already exist".format(
+                    hive_table_model.tracking_table_name), logging.CRITICAL)
+                return False
+            if self.check_bq_table_exists(bq_table_model.dataset_id,
+                                          bq_table_model.table_name):
+                print_and_log(
```

Instead of returning False and logging this message raise `google.api_core.exceptions.AlreadyExists`